docs/ebpf/guides/getting-started.md

```diff
@@ -0,0 +1,311 @@
+# Creating an eBPF-powered Go application
+
+!!! tip ""
+    For a high-level understanding of what eBPF is and how it works, please see
+    [the eBPF introduction at ebpf.io](https://ebpf.io/what-is-ebpf). This guide
+    assumes familiarity with the basic concepts and terminology, as well as a
+    basic understanding of the Go toolchain (`go build`). We'll also rely on `go
```

Comment:
I think we should go much further and assume that the reader is familiar with the `go` language itself and knows how to write idiomatic code. There are plenty of resources out there for people that are not yet at that level.